packages/eslint/src/generators/lint-project/lint-project.ts

```diff
@@ -71,19 +59,29 @@ export async function lintProjectGenerator(
   });
   const projectConfig = readProjectConfiguration(tree, options.project);
 
-  const lintFilePatterns = options.eslintFilePatterns ?? [];
-  if (isBuildableLibraryProject(projectConfig)) {
-    lintFilePatterns.push(`${projectConfig.root}/package.json`);
-  }
-
   projectConfig.targets['lint'] = {
     executor: '@nx/eslint:lint',
     outputs: ['{options.outputFile}'],
-    options: {
-      lintFilePatterns: lintFilePatterns,
-    },
   };
 
+  let lintFilePatterns = options.eslintFilePatterns;
+  if (!lintFilePatterns && options.rootProject) {
+    lintFilePatterns = ['./src'];
```

Comment:
Ok so this is the answer to my previous question... but is this correct?

In the case of other projects we lint other files like the package.json and project.json by default, but this wouldn't do that?

Reply:
This is how it has been working so far. You can always override this behavior by explicitly including those files via `lintFilePatterns`. Unfortunately, it's the limitation of the standalone projects.

Reply:
But we could include project.json and package.json explicitly in the patterns couldn't we?

Reply:
I understand not including all files, but we could include ones explicitly that we know to be important?

Reply:
This is the default generator that never includes the `package.json` by default unless it's a buildable lib (which case is handled underneath on line 76). 

Users can always subsequently add any paths they want, and this only simplifies the default generation.

Reply:
Moreover, having a standalone buildable library with `@nx/dependency-checks` (use case where we do want to test package.json) does not make too much sense since the root `package.json` would have to include a lot of additional dependencies which would not be part of the package so there would have to be a lot of `ignores` set in place.

Reply:
Ok sounds good, resolved 👍